mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -105,21 +112,33 @@ def __init__(self, artifact_uri):
             get_databricks_profile_uri_from_artifact_uri(artifact_uri)
             or mlflow.tracking.get_tracking_uri()
         )
-        self.run_id = self._extract_run_id(self.artifact_uri)
-        # Fetch the artifact root for the MLflow Run associated with `artifact_uri` and compute
-        # the path of `artifact_uri` relative to the MLflow Run's artifact root
-        # (the `run_relative_artifact_repo_root_path`). All operations performed on this artifact
-        # repository will be performed relative to this computed location
-        artifact_repo_root_path = extract_and_normalize_path(artifact_uri)
-        run_artifact_root_uri = self._get_run_artifact_root(self.run_id)
-        run_artifact_root_path = extract_and_normalize_path(run_artifact_root_uri)
-        run_relative_root_path = posixpath.relpath(
-            path=artifact_repo_root_path, start=run_artifact_root_path
-        )
-        # If the paths are equal, then use empty string over "./" for ListArtifact compatibility.
-        self.run_relative_artifact_repo_root_path = (
-            "" if run_artifact_root_path == artifact_repo_root_path else run_relative_root_path
-        )
+
+    @property
+    def run_id(self):
+        if self._run_id is None:
+            self._run_id = self._extract_run_id(self.artifact_uri)
+
+        return self._run_id
+
+    @property
+    def run_relative_artifact_repo_root_path(self):
+        if self._run_relative_artifact_repo_root_path is None:
+            # Fetch the artifact root for the MLflow Run associated with `artifact_uri` and compute
+            # the path of `artifact_uri` relative to the MLflow Run's artifact root
+            # (the `run_relative_artifact_repo_root_path`). All operations performed on this
+            # artifact repository will be performed relative to this computed location
+            artifact_repo_root_path = extract_and_normalize_path(self.artifact_uri)
+            run_artifact_root_uri = self._get_run_artifact_root(self.run_id)
+            run_artifact_root_path = extract_and_normalize_path(run_artifact_root_uri)
+            run_relative_root_path = posixpath.relpath(
+                path=artifact_repo_root_path, start=run_artifact_root_path
+            )
+            # If the paths are equal, then use empty string over "./" for ListArtifact compatibility
+            self._run_relative_artifact_repo_root_path = (
+                "" if run_artifact_root_path == artifact_repo_root_path else run_relative_root_path
+            )
+
+        return self._run_relative_artifact_repo_root_path
```

Comment:
This is to avoid fetching a run and hitting a `NotFound` error.

Reply:
The previous behavior set the `self.run_relative_artifact_repo_root_path` to `""` at init time. 
This is a behavior change, right?

Reply:
Now we're initializing this value as soon as it's used, as opposed to during `__init__`. Seems fairly reasonable to me.